editor.planx.uk/src/@planx/components/Page/Public.tsx

```diff
@@ -1,19 +1,59 @@
 import { PublicProps } from "@planx/components/ui";
+import { FormikConfig, useFormik } from "formik";
 import React from "react";
 
+import { flatten } from "../List/utils";
 import Card from "../shared/Preview/Card";
 import CardHeader from "../shared/Preview/CardHeader";
+import { useSchema } from "../shared/Schema/hook";
+import { SchemaUserData } from "../shared/Schema/model";
+import { SchemaFields } from "../shared/Schema/SchemaFields";
+import { getPreviouslySubmittedData } from "../shared/utils";
 import { Page } from "./model";
 
 type Props = PublicProps<Page>;
 
 function PageComponent(props: Props) {
+  const { formikConfig } = useSchema({
+    schema: props.schema,
+    previousValues: getPreviouslySubmittedData(props),
+  });
+
+  const onSubmit: FormikConfig<SchemaUserData>["onSubmit"] = (data) => {
```

Comment:
As with the `List` component, we're preversing both forms of data here as opposed to writing to and parsing from a single format.

This seems like an ok approach, especially as it's following an established pattern?

Reply:
By "preversing both forms of data" you're referring to `defaultPassportData` and `flattenedData`?

Reply:
Yes, that's right. We could parse/map/serialise both into a single structure and then unpack that on "back" navigation, but I'm proposing we allow both just now 👍